### PR TITLE
display a short documentation with the -l/ show envlist (a concatenation of the environments description) 

### PR DESCRIPTION
**Pull request by gaborbernat**
This is the implementation for https://github.com/tox-dev/tox/issues/462. 

For example:
```ini
[tox]
envlist={py27,py36}-{windows,linux}
[testenv]
description= py27: run py.test on Python 2.7
             py36: run py.test on Python 3.6
             windows: on Windows platform
             linux: on Linux platform
             docs: generate documentation
commands=py.test {posargs}

[testenv:docs]
changedir = docs
```

```bash
tox -l 
py27-windows run py.test on Python 2.7 on Windows platform
py27-linux   run py.test on Python 2.7 on Linux platform
py36-windows run py.test on Python 3.6 on Windows platform
py36-linux   run py.test on Python 3.6 on Linux platform
```

### CODE REVIEW

File: tox/session.py [hunk 1]
```diff
@@ -627,8 +627,9 @@ def showconfig(self):
                                  % (attr.name, getattr(envconfig, attr.name)))
 
     def showenvs(self):
+        max_length = max(len(env) for env in self.config.envlist)
         for env in self.config.envlist:
-            self.report.line("%s" % env)
+            self.report.line("{} {}".format(env.ljust(max_length), self.config.envconfigs[env].description))
```
Comment (The-Compiler) [inline tox/session.py]: This should be `{0} {1}` to work on Python 2.6 - it seems like it doesn't get caught in the tests though, and they report "can not run tests involving calling tox on python2.6. (and python2.6 is about to be deprecated anyway)"...

Comment (gaborbernat) [inline tox/session.py]: @The-Compiler indeed, fixed that, thanks!

**Comment by gaborbernat** [comment]
@The-Compiler , @obestwalter now that the CI checks passed, can you guys take a look? thanks!

**Comment by The-Compiler** [comment]
LGTM (though I don't know tox' code very well), but I'll let @obestwalter press the button :wink:

**Comment by nicoddemus** [comment]
LGTM as well, but I feel we should add docs about this new feature first.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
+.vagrant
```
Comment (obestwalter) [inline .gitignore]: ha - nice to see it's being used already - does not really belong to this PR, but won't hurt anyone to just add it now, I guess :)

**Comment by obestwalter** [comment]
@The-Compiler what an honour :D - I agree with  @nicoddemus that this deserves some documentation (somewhere [here-ish](http://tox.readthedocs.io/en/latest/config.html)) before we call this done.

**Comment by gaborbernat** [comment]
@obestwalter hmm, I thought that's automatically taken during documentation generation from here:

```
 parser.add_testenv_attribute(
        name="description", type="string", default='', postprocess=merge_description,
        help="short description of this environment")
```

Where is that shown otherwise?

**Comment by The-Compiler** [comment]
When doing `tox --help`

**Comment by gaborbernat** [comment]
@obestwalter , @The-Compiler done :+1:

**Comment by obestwalter** [comment]
Played around with it locally a bit and it looks good. Looking forward to the next step that fixes #271.

**Comment by The-Compiler** [comment]
I can't get this to work with the 2.7.0 preview with [this tox.ini](https://github.com/qutebrowser/qutebrowser/blob/master/tox.ini). Both `description = Hello World` in `[testenv:py36-cov]` (which is what I'd expect to work by reading the docs) and `description = py36-cov: Hello World` in `[testenv]` (which seems odd, but what the tests do) still gives me no description in the env list.

**Comment by obestwalter** [comment]
o.k. just tried the unconditional approach with 2.7.0 from devpi with this ini:

    [tox]
    envlist={py27,py36}-{windows,linux}
    [testenv]
    description= hello world

    [testenv:docs]
    description = blablabla


     $ tox -l

    py27-windows hello world
    py27-linux   hello world
    py36-windows hello world
    py36-linux   hello world

    $ tox -l -e docs
    docs blablabla

So from how I understand it, it works like ti should. Am I missing something?

**Comment by The-Compiler** [comment]
Oh, damn, my bad! I was running my systemwide tox instead of the one from the virtualenv... nevermind! 😊

**Comment by gaborbernat** [comment]
@obestwalter cool, submitted the PR for #271 in #495
